libindy/src/commands/metrics.rs

```diff
@@ -55,26 +56,47 @@ impl MetricsCommandExecutor {
     }
 
     fn append_threapool_metrics(&self, metrics_map: &mut Map<String, Value>) {
+        struct MetricsTags {
+            stage: String,
+        }
         let tp_instance = crate::commands::THREADPOOL.lock().unwrap();
-        metrics_map.insert(String::from(THREADPOOL_ACTIVE_COUNT),
-                           Value::from(tp_instance.active_count()));
-        metrics_map.insert(String::from(THREADPOOL_QUEUED_COUNT),
-                           Value::from(tp_instance.queued_count()));
-        metrics_map.insert(String::from(THREADPOOL_MAX_COUNT),
-                           Value::from(tp_instance.max_count()));
-        metrics_map.insert(String::from(THREADPOOL_PANIC_COUNT),
-                           Value::from(tp_instance.panic_count()));
+        let metrics_map = checkMetricsStructure(metrics_map);
+        let mut threadpool_threads_count = Vec::new();
+
+        threadpool_threads_count.push(Value::from("value", tp_instance.active_count()));
+        threadpool_threads_count.push(Value::from("tags", MetricsTags{stage: "active".to_string()}));
+
+        threadpool_threads_count.push(Value::from("value", tp_instance.queued_count()));
+        threadpool_threads_count.push(Value::from("tags", MetricsTags{stage: "queued".to_string()}));
+
+        threadpool_threads_count.push(Value::from("value", tp_instance.max_count()));
+        threadpool_threads_count.push(Value::from("tags", MetricsTags{stage: "max".to_string()}));
```

Comment:
`max` state for thread looks like a.. stretch :) Why not a separate metric in this case?